commec/regulation/region.py

```diff
@@ -0,0 +1,148 @@
+
+"""
+The Region module handles the sanitization, and import, of counties
+and regions. Supports all current alpha2 country codes through pycountry
+module. However also overrides this behaviour with an imported region_definitions.json
+data file, which supports general region nesting, such as EU, for the European Union.
+
+We require this distinction because of the nature of political and policy making
+entities not being a true 1:1 with countries. Furthermore, we want the user to input
+countries they are interested in from a regulation standpoint, and then get relevant
+commec outputs based on this choice.
+
+Someone from Germany would be interested in the policies made that affect
+the European Union for example.
+"""
+
+import os
+import json
+import logging
+import pycountry as pc
+
+from commec.regulation.containers import Region
+
+logger = logging.getLogger(__name__)
+
+REGION_DATA_LUT = {}
+
+def load_region_list_data(input_filepath : str | os.PathLike):
+    """
+    Load the region to country data based on input filepath.
+    Input filepath contains a json list containing mapping of 
+    two letter codes and names, to a list of affected regions.
+    """
+
+    if not os.path.isfile(input_filepath):
+        logger.warning("No additional region definitions found at expected location: %s",
+        input_filepath)
+        return
+
+    region_array = []
+    global REGION_DATA_LUT
```

Comment:
having an itchy feeling of "why globals... maybe should have classes" but maybe I am too object-oriented